### PR TITLE
Add Redshift Adapter

### PR DESCRIPTION
**Pull request by brettshollenberger**
* ✨  Adds support for Redshift ✨ 
* Run tests with `REDSHIFT_URL=redshift_url rake test:redshift`

### CODE REVIEW

File: minitest/redshift_plugin.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require "minitest"
+
+module Minitest
+  def self.plugin_redshift_options(opts, options)
+    opts.on "--redshift REDSHIFT_URL", "Configure Redshift connection string" do |redshift_url|
```
Comment (ankane) [inline minitest/redshift_plugin.rb]: Can we use `ENV["REDSHIFT_URL"]` instead, which would allow us to remove this file?

**Comment by brettshollenberger** [comment]
@ankane done; tests run now with `REDSHIFT_URL=redshift_url rake test:redshift`

File: test/test_helper.rb [hunk 2]
```diff
@@ -335,6 +351,9 @@ def create_user(created_at, score = 1)
     # hack for MySQL adapter
     # user.update_attributes(created_at: nil, created_on: nil) if created_at.nil?
 
+    # hack for Redshift adapter, which doesn't return id on creation...
+    user = User.last
```
Comment (ankane) [inline test/test_helper.rb]: Can we only do this if `id` doesn't get set?

File: test/redshift_test.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+require_relative "test_helper"
+
+class TestRedshift < Minitest::Test
+  include TestGroupdate
+  include TestDatabase
+
+  def setup
+    super
+    @@setup ||= begin
+      if ENV["REDSHIFT_URL"].present?
```
Comment (ankane) [inline test/redshift_test.rb]: What if we had an error about needing to set `REDSHIFT_URL` instead?

**Comment by ankane** [comment]
Looks like CI is failing. We'll need to remove the Redshift tests from `rake test` in the `Rakefile`.

**Comment by ankane** [comment]
Also, looks like a gem compatibility issue.  Instead of adding the redshift adapter to dev dependencies, I think we'll need to specify it in the `Gemfile`.

File: test/redshift_test.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+require_relative "test_helper"
+
+class TestRedshift < Minitest::Test
+  include TestGroupdate
+  include TestDatabase
+
+  def setup
+    super
+    @@setup ||= begin
+      abort("REDSHIFT_URL environment variable must be set in order to run tests") unless ENV["REDSHIFT_URL"].present?
```
Comment (brettshollenberger) [inline test/redshift_test.rb]: @ankane had to resort to abort to get Minitest not to run the tests anyway, but it's much cleaner now

**Comment by brettshollenberger** [comment]
I don't think we should need the adapter to be in the main Gemfile; I think the compatibility issue will be resolved by explicitly specifying the version dependency of ActiveRecord in dev dependencies (did that now).

Also moved the tests to `rake noncompulsory_tests:redshift`, so I think we should move past these build errors now.

**Comment by ankane** [comment]
Not sure that'll solve it. https://travis-ci.org/ankane/groupdate/builds/141497622

File: test/redshift_test.rb (hunk 4 above)
Comment (ankane) [inline test/redshift_test.rb]: 👍

**Comment by brettshollenberger** [comment]
Yea..

**Comment by ankane** [comment]
Tests green :) Getting close. Added a few more comments inline.

File: groupdate.gemspec [hunk 5]
```diff
@@ -23,7 +23,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "bundler", "~> 1.3"
   spec.add_development_dependency "rake"
   spec.add_development_dependency "minitest"
-  spec.add_development_dependency "activerecord"
```
Comment (ankane) [inline groupdate.gemspec]: Any reason for this?

File: test/noncompulsory/redshift_test.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+class TestRedshift < Minitest::Test
+  include TestGroupdate
+  include TestDatabase
+
+  def setup
+    super
+    @@setup ||= begin
+      abort("REDSHIFT_URL environment variable must be set in order to run tests") unless ENV["REDSHIFT_URL"].present?
+
+      uri      = URI.parse(ENV["REDSHIFT_URL"])
```
Comment (ankane) [inline test/noncompulsory/redshift_test.rb]: Instead of parsing the url, can we use ActiveRecord's `url` option?

File: Rakefile [hunk 7]
```diff
@@ -17,3 +17,10 @@ namespace :test do
     t.pattern = "test/mysql_test.rb"
   end
 end
+
+namespace :noncompulsory_tests do
```
Comment (ankane) [inline Rakefile]: I like the consistency of `test:redshift` with `test:postgresql` and `test:mysql`. Maybe we could keep `test:redshift` and modify `test` to use the `test_files` option of Rake::TestTask. http://ruby-doc.org/stdlib-2.0.0/libdoc/rake/rdoc/Rake/TestTask.html

File: groupdate.gemspec (hunk 5 above)
Comment (brettshollenberger) [inline groupdate.gemspec]: We also have it in dependencies; not a strong reason, just duplicative :)

Comment (ankane) [inline groupdate.gemspec]: `activesupport` is in dependencies, so would prefer to keep it.

File: Rakefile [hunk 8]
```diff
@@ -4,7 +4,7 @@ require "rake/testtask"
 task default: :test
 Rake::TestTask.new do |t|
   t.libs << "test"
-  t.pattern = "test/**/*_test.rb"
+  t.test_files = FileList["test/**/*_test.rb"].exclude(/redshift/)
```
Comment (ankane) [inline Rakefile]: Nice!

**Comment by brettshollenberger** [comment]
Ah whoops, you're right; cool everything looks right to me now then 🚢

**Comment by ankane** [comment]
Beautiful 💯 Thanks Brett!

**Comment by brettshollenberger** [comment]
@ankane hold up strange error on that build

**Pull request merged**